libs/telemetry/src/main/java/org/opensearch/telemetry/tracing/WrappedSpan.java

```diff
@@ -0,0 +1,29 @@
+/*
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * The OpenSearch Contributors require contributions made to
+ * this file be licensed under the Apache-2.0 license or a
+ * compatible open source license.
+ */
+
+package org.opensearch.telemetry.tracing;
+
+/**
+ * Wrapped Span will be exposed to the code outside of tracing package for sharing the {@link Span} without having access to
+ * its properties.
+ */
+public final class WrappedSpan {
```

Comment:
Why do we need to wrap `Span` and not use `Span` directly? This additional layer is not needed I think

Reply:
Only reason is, Span will have access to all the operations like addAttribute, endSpan, etc.

Reply:
Oh see, we basically model over the `Context` (from Otel), I think the `WrappedSpan` is not a good name for it, may be `SpanContext` or something? (we should also allow to manage attributes since it is very common need to enrich current context with attributes but we could do that later)

Reply:
Make sense, Let me do that. I will extract these addAttribute, addEvent methods to some interface and let all Span, SpanScope and SpanContext implement this interface. Span will have endSpan method additionally.

Reply:
Now, getCurrentSpan can return the MinimalSpan only but still somebody can cast is back to Span and call endSpan method. So we need to wrap it in SpanContext and return

Reply:
Yes agreed, only point is since all the classes Span, SpanScope and eventually SpanContext will be having the same methods for add event and attribute. So any new method addition would required to be added at all the places. So was just thinking if we can extract out the definition in an interface which all these classes can implement.

Reply:
I think we could always revisit that later, adding new public API is much easier than removing